php-classes/Slate/CBL/StudentCompetency.php

```diff
@@ -287,16 +289,29 @@ public function getDemonstrationOpportunities()
     {
         if ($this->demonstrationOpportunities === null) {
             $this->demonstrationOpportunities = 0;
-
+            $hasWildCard = false;
             foreach ($this->getDemonstrationsData() as $skillId => $demonstrationData) {
                 foreach ($demonstrationData as $demonstration) {
-                    if (empty($demonstration['Override'])) {
-                        $this->demonstrationOpportunities++;
+                    // skip overrides by class
+                    if ($demonstration['DemonstrationClass'] == OverrideDemonstration::class) {
+                        continue;
+                    }
+
+                    if ($demonstration['EvidenceWeight'] === null) {
+                        $hasWildcard = true;
+                        break;
+                    } else {
+                        $this->demonstrationOpportunities += $demonstration['EvidenceWeight'];
                     }
                 }
             }
         }
 
+        if ($hasWildCard) {
+            // return total requirements
+            $this->demonstrationOpportunities = $this->getDemonstrationsRequired();
+        }
```

Comment:
Without this, could `$this->demonstrationOpportunities` ever be greater than `$this->getDemonstrationsRequired();` ? should this block handle capping it too if EvidenceWeight values > 1 are able to produce a higher count than required?

Reply:
This doesn't look like it was previously working at all due to case mismatch in `$hasWildcard` vs `$hasWildCard`

Reply:
fixed in d2b35a077f6be09cc64f996ee7ad97c326cc2093